src/Orleans.Serialization/Serializers/CodecProvider.cs

```diff
@@ -169,39 +169,50 @@ public IFieldCodec TryGetCodec(Type fieldType)
             // If the field type is unavailable, return the void codec which can at least handle references.
             return fieldType is null ? _voidCodec
                 : _untypedCodecs.TryGetValue(fieldType, out var existing) ? existing
-                : TryCreateCodec(fieldType);
-        }
+                : GetOrCreateCodec(fieldType);
```

Comment:
This should be just `TryCreateCopier(fieldType) is { } res ? _untypedCopiers.GetOrAdd(fieldType, res) : null;` like in `TryGetDeepCopier` (where I should have copied this line, but somehow forgot 🤦‍♂️).